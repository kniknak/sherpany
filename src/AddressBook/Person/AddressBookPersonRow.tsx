import * as React from "react";
import TableCell from "@material-ui/core/TableCell";
import { FetchAddressBookUser } from "../redux/types";
import MoreIcon from "@material-ui/icons/MoreVert";
import IconButton from "@material-ui/core/IconButton";
import Avatar from "@material-ui/core/Avatar";

export interface AddressBookPersonRowProps {
    person: FetchAddressBookUser
    onOpen: () => void
}

export class AddressBookPersonRow extends React.PureComponent<AddressBookPersonRowProps> {
    render() {
        const { person, onOpen } = this.props;

        return (
            <>
                <TableCell>
                    <Avatar
                        alt={person.name.first + " " + person.name.last}
                        src={person.picture.thumbnail}
                    />
                </TableCell>
                <TableCell>{person.name.first}</TableCell>
                <TableCell>{person.name.last}</TableCell>
                <TableCell>{person.login.username}</TableCell>
                <TableCell>{person.email}</TableCell>
                <TableCell>
                    <IconButton
                        color="inherit"
                        onClick={onOpen}
                    >
                        <MoreIcon/>
                    </IconButton>
                </TableCell>
            </>
        );
    }
}