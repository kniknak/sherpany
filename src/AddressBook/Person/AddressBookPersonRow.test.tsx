import React from "react";
import { shallow } from "enzyme";
import { AddressBookPersonRow, AddressBookPersonRowProps } from "./AddressBookPersonRow";

describe("AddressBook", () => {
    describe("Person", () => {
        describe("AddressBookPersonRow", () => {
            it("renders", () => {
                const props: AddressBookPersonRowProps = {
                    person: {
                        name: {
                            first: "name.first",
                            last: "name.last",
                        },
                        location: {
                            street: {
                                name: "location.street.name",
                                number: 1,
                            },
                            city: "location.city",
                            state: "location.state",
                            postcode: "location.postcode",
                        },
                        email: "email",
                        login: {
                            username: "login.username",
                        },
                        phone: "phone",
                        cell: "cell",
                        id: {
                            value: "id.value",
                        },
                        picture: {
                            thumbnail: "picture.thumbnail",
                        },
                    },
                    onOpen: () => undefined,
                };

                const subject = shallow(<AddressBookPersonRow {...props}/>);

                expect(subject).toMatchSnapshot()
            });
        });
    });
});