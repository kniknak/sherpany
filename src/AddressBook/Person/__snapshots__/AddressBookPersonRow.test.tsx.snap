// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressBook Person AddressBookPersonRow renders 1`] = `
<Fragment>
  <WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(Avatar))
      alt="name.first name.last"
      src="picture.thumbnail"
    />
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    name.first
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    name.last
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    login.username
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    email
  </WithStyles(ForwardRef(TableCell))>
  <WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(IconButton))
      color="inherit"
      onClick={[Function]}
    >
      <MoreVertIcon />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(TableCell))>
</Fragment>
`;
